src/TortoiseProc/DiffLinesForStaging.cpp

```diff
@@ -0,0 +1,238 @@
+﻿// TortoiseGit - a Windows shell extension for easy version control
+
+// Copyright (C) 2016, 2019-2020 - TortoiseGit
+// Copyright (C) 2007, 2009-2013, 2020 - TortoiseSVN
+
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software Foundation,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+#include "stdafx.h"
+#include "DiffLinesForStaging.h"
+#include <regex>
+
+CDiffLinesForStaging::CDiffLinesForStaging(const std::unique_ptr<char[]>& text, int numLines, int firstLineSelected, int lastLineSelected)

```

Comment:
Please don't use unique_ptr<>&, better directly use char* here

Reply:
You got me wrong here. Just use `char*` here and don't pass a const reference to a `unique_ptr`.